libsolidity/ast/Types.cpp

```diff
@@ -575,11 +575,39 @@ tuple<bool, rational> RationalNumberType::isValidLiteral(Literal const& _literal
 	rational x;
 	try
 	{
-		rational numerator;
-		rational denominator(1);
-		
+		auto expPoint = find(_literal.value().begin(), _literal.value().end(), 'e');
+		if (expPoint == _literal.value().end())
+			expPoint = find(_literal.value().begin(), _literal.value().end(), 'E');
 		auto radixPoint = find(_literal.value().begin(), _literal.value().end(), '.');
-		if (radixPoint != _literal.value().end())
+
+		if (expPoint != _literal.value().end())
```

Comment:
This all got quite complex now. I wonder whether it might make sense to use regular expressions instead.